ejb3/src/main/java/org/jboss/as/ejb3/tx/CMTTxInterceptor.java

```diff
@@ -79,7 +79,7 @@
      */
     protected void endTransaction(final TransactionManager tm, final Transaction tx) {
         try {
-            if (tx != tm.getTransaction()) {
+            if (! tx.equals(tm.getTransaction())) {
```

Comment:
Excellent, thanks for the explanation!  With the change to use "tx.equals(tm.getTransaction())", we should be more robust in the future, as we will no longer throw an exception and leave the transaction associated with the thread.